engines/zypper_auth/lib/zypper_auth/engine.rb

```diff
@@ -27,12 +27,21 @@ def verify_instance(request, logger, system)
       Rails.cache.write(cache_key, is_valid, expires_in: 20.minutes)
       is_valid
     rescue InstanceVerification::Exception => e
+      message = ''
+      if system.proxy_byos
+        result = SccProxy.scc_check_subscription_expiration(request.headers, system.login, logger) if system.proxy_byos
+        return true if result[:is_active]
```

Comment:
This is not getting put in the cache, so would the check get performed on each zypper access to the repository? That can get pretty slow.

Reply:
In case the subscription is active, cache gets updated, other than that, yes system subscriptions needs to get checked